supervise.py

```diff
@@ -267,9 +267,13 @@ def stop(self):
         self._signal("d")
 
     def _signal(self, signal):
-        f = open(self._control,"w+")
-        f.write(signal)
-        f.close()
+        try:
+            f = open(self._control,"w+")
+            f.write(signal)
+        except IOError as e:
+            raise ServiceControlError("Unable to write service control file '{0}'".format(e.filename))
+        else:
```

Comment:
I was thinking:
```
f = None
try:
    f = open(...)
    f.write(...)
except IOError as e:
    raise SuperviseControlError(...)
finally:
    if f is not None:
        f.close()
```

Splitting and the extra try around the close is probably overkill, but I'm not used to weird scenarios that you can't close fds on.